src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else if (size == 1) {
+            Node newNode = new Node(head, value, null);
+            head.next = newNode;
+            tail = newNode;
+        } else {
+            Node newNode = new Node(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        indexCheckForAdd(index);
+        Node indexNode = indexOf(index);
+        if (indexNode == null) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            Node newNode = new Node(null, value, head);
+            head.prev = newNode;
+            head = newNode;
+            size++;
+            return;
+        }
+        Node newNode = new Node(indexNode.prev, value, indexNode);
+        indexNode.prev.next = newNode;
+        indexNode.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
```

Comment:
```suggestion
        for (T element : list) {
```
Use understandable naming